docs/Architecture.md

```diff
@@ -96,33 +76,208 @@ queue
+    Worker->>Solr: add(document)
+    Worker->>Postgres: update('success')
+    Worker->>CompletedQueue: QueueEntry
+    
+    CompletedQueue->>Controller: handleCompleted(success)
+    Controller->>CompletedQueue: acknowledgement
+  else failure
+    Worker->>Postgres: update('failure')
+    Worker->>CompletedQueue: QueueEntry
+    CompletedQueue->>Controller: handleCompleted(failure)
+    Controller->>CompletedQueue: acknowledgement
+  end
+```
 
 * This sequence diagram shows how a trigger event on a MN or CN results in a job being added to the pending queue.
 
-![Queue Event Sequence](https://github.com/NCEAS/metadig-engine/blob/master/docs/images/queue-event-trigger_sequence.png "Queue Event Sequence")
+```mermaid
+sequenceDiagram
+  participant D1Client
+  participant Metacat
+  participant Scheduler
+  participant Controller
+  participant Queue
+  participant Worker
+
+  title MetaDIG Engine: 'Quality' Queue
+
+  activate D1Client
+  D1Client ->> Metacat: MN.create(metadata, sysmeta)
+  deactivate D1Client
+
+  activate Metacat
+  Metacat --> Scheduler: Harvest
+  deactivate Metacat
+
+  activate Scheduler
+  Scheduler ->> Scheduler: MDQClient.requestReport(sysmeta)
+  Scheduler ->> Controller: https://quality..../suite/{id}/run
+  deactivate Scheduler
+
+  activate Controller
+  Controller->>Queue: BasicPublish
+  deactivate Controller
+
+  Queue ->> Worker: BasicConsume
+  activate Worker
+  Worker ->> Worker: processRun(metadata, sysmeta, suiteId)
+  deactivate Worker
+```
 
 ## The metadig-engine scheduler
 
 * This sequence diagram shows how the scheduler determines which DataONE metadata documents to create quality reports for, then submits requests to metadig-engine to create the reports.
 
-![Scheduler Sequence](https://github.com/NCEAS/metadig-engine/blob/master/docs/images/index-monitor_sequence.png "Scheduler Sequence")
+```mermaid
+sequenceDiagram
+  participant Scheduler
+  participant MNIndex
+  participant QualityIndex
+  participant Controller
+
+  title MetaDIG Engine: Index Monitor
+
+  activate Scheduler
+  Scheduler ->> MNIndex: query(pid="*", updateId)
+  activate MNIndex
+  MNIndex ->> Scheduler: pid list
+  deactivate MNIndex
+  Scheduler ->> QualityIndex: query(pid, suiteId)
+  activate QualityIndex
+  QualityIndex ->> Scheduler: pid status
+  alt does not exist
+    Scheduler ->> Controller: /suite/id/run
+    Controller ->> QualityIndex: add()
+  end
+  deactivate QualityIndex
+  deactivate Scheduler
+```
+
+### Monitoring for stuck jobs
+
+The controller launches a quartz job on a schedule configurable in the metadig.properties file to make sure that quality jobs don't get "stuck" in the processing state. This might happen if a worker unexpectedly dies mid-process, after acknowledging the RabbitMQ quality message, but before completing the task. The following sequence diagram shows how this process works. In this case the "Client" could either be a direct request from the API or (more likely) the scheduler.
```

Comment:
Glad to have this section. It would be useful to more fully explain the processing queue logic and sequence, including how we came to using two queues because of Metadig timeout issues. It would be nice for a developer to understand the logic that led us here, and to be able to understand how your resubmission system works.